evaluate/repository.go

```diff
@@ -86,6 +87,26 @@ func TemporaryRepository(logger *log.Logger, dataPath string) (temporaryReposito
 		return "", cleanup, pkgerrors.WithStack(err)
 	}
 
+	// Add a default git configuration.
+	if err := os.MkdirAll(filepath.Join(temporaryRepositoryPath, ".git"), 0700); err != nil {
+		return "", cleanup, pkgerrors.WithStack(err)
+	}
+	if err := os.WriteFile(filepath.Join(temporaryRepositoryPath, ".git", "config"), bytesutil.TrimIndentations([]byte(`
```

Comment:
Instead of writing a file I would simply pass `GIT_CONFIG_NOSYSTEM` https://git-scm.com/docs/git-config#Documentation/git-config.txt-GITCONFIGNOSYSTEM as environement variable to all `CommandWithResult` calls and then set the `dummy-name` stuff with `git config user.name "Your Name Here"`

Reply:
Or at least run `git init` with `GIT_CONFIG_NOSYSTEM` and then simply override the `.git/config` file with the dummy values.

Reply:
But I think even `GIT_CONFIG_NOSYSTEM` only disables the system config but then the global user config would still be read, right? And changing only the username does not i.e. reset the GPG signing to the default disabled state then.

So we could run `git init` and then `git config user.name ...` and `git config user.mail ...` though that is two subprocess spawns extra vs. one file operation, that we run for each repository setup?

Reply:
its not worth it if it is working.